crates/sui-framework/sources/safe.move

```diff
@@ -0,0 +1,214 @@
+/// The Safe standard is a minimalistic shared wrapper around a coin. It provides a way for users to provide third-party dApps with
+/// the capability to transfer coins away from their wallets, if they are provided with the correct permission.
+module sui::safe {
+    use sui::object::{Self, ID, UID};
+    use sui::tx_context::{TxContext, sender};
+    use sui::transfer::Self;
+    use sui::balance::{Self, Balance};
+    use std::vector::Self;
+
+    ///
+    /// Allows any holder of a capability to transfer a fixed amount of assets from the safe.
+    /// Useful in situations like an NFT marketplace where you wish to buy the NFTs at a specific price.
+    /// 
+    /// @ownership: Shared
+    /// 
+    struct Safe<phantom T> has key {
+        id: UID,
+        owner: address,
+        balance: Balance<T>,
+        permissions_bitmap: vector<u64>,
+        next_nonce: u64,
+    }
+
+    ///
+    /// Allows the owner of the capability to take `amount` of coins from the box.
+    ///
+    /// @ownership: Owned
+    ///
+    struct Capability<phantom T> has store, key {
```

Comment:
If we call this something like `AmountCap`, that could open the door for issuing other kinds of capabilities in the future (e.g., a cap that gives you ownership of a share of the funds rather than a fixed amount). Maybe that wants to be a different standard, but could be an interesting building block.

Unrelated observation: a `FlashLender` does not look so different than a `Safe`--just a different revocability policy (you must self-revoke by end of tx + pay to do so) https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/defi/sources/flash_lender.move#L13